pennylane/math/utils.py

```diff
@@ -169,7 +172,66 @@ def convert_like(tensor1, tensor2):
     return np.asarray(tensor1, like=interface)
 
 
-def get_interface(tensor):
+def get_interface(*values):
+    """Determines the correct framework to dispatch to given a tensor or a
+    sequence of tensor-like objects.
+
+    Args:
+        values (tensor_like, Sequence[tensor_like]): a sequence of tensor like objects
+
+    Returns:
+        str: the name of the interface
+
+    To determine the framework to dispatch to, the following rules
+    are applied:
+
+    * Tensors that are incompatible (such as Torch, TensorFlow and Jax tensors)
+      cannot both be present.
+
+    * Autograd tensors *may* be present alongside Torch, TensorFlow and Jax tensors,
+      but Torch, TensorFlow and Jax take precendence; the autograd arrays will
+      be treated as non-differentiable NumPy arrays. A warning will be raised
+      suggesting that vanilla NumPy be used instead.
+
+    * Vanilla NumPy arrays and SciPy sparse matrices can be used alongside other tensor objects;
+      they will always be treated as non-differentiable constants.
+    """
+
+    if len(values) == 1:

```

Comment:
> Note: Careful, because qml.math.get_interface will always return "numpy" when calling it with a list of tensors, given that it will assume that the user wants to get the interface of the list, and get_interface always returns "numpy" for Python built-ins:

Was this the behaviour before for `get_interface`? I fear that this can be forgotten by developers/users, could it make sense to raise a warning? Not sure if getting the interface of a Pyhton built-in is a frequent use case.

Reply:
It was! Given that before `get_interface` could only be called for one input, if you used a list it returned "numpy". 

I mentioned this because this was NOT the case for the `_multi_dispatch` function. However since this was a private function I don't think it is a big problem.

Reply:
Could we clear up this behaviour in some way? The argument section of the docstring includes `Sequence[tensor_like]`, which suggests that for example a list of tensor-like objects is accepted. Although such a list is accepted indeed, the result of the function for such an argument doesn't have any practical usage.

Adding a note/warning box describing the behaviour of passing in a sequence of tensor-like objects could be a potential approach.